core/src/Revolution/Sources/modMediaSource.php

```diff
@@ -2087,13 +2087,19 @@ protected function getExtJSDirClasses()
      *
      * @return array
      */
-    protected function getListDirContextMenu()
+    protected function getListDirContextMenu($dirName)
```

Comment:
Any chance we can accomplish this without changing the function signature? As it's not a private method, this can cause issues in extended classes.

Reply:
In my opinion, it is impossible to do without changing the signature.
Alternatively, we can set a default value:

```
protected function getListDirContextMenu($dirName = null) {
    ...

    $menu = [];
    if (isset($dirName)) {
        $menu[] = [
            'text' => '<b>' . htmlentities($dirName) . '</b>',
            'handler' => '',
            'header' => true,
        ];
        $menu[] = '-';
    }
    ...

    return $menu;
}
```

Reply:
What about adding it client-side, like you did in modx.tree.directory.js?

Reply:
You are right, I found a simpler solution, and without editing the signature.